src/bda/plone/discount/browser/forms.py

```diff
@@ -206,6 +212,15 @@ class GroupDiscountFormBase(DiscountFormBase):
     for_callback = 'javascript:discount_form.autocomplete_group'
     for_mode = 'edit'
 
+class CouponDiscountFormBase(DiscountFormBase):
+    header_template = 'coupon_header.pt'
+    for_attribute = FOR_COUPON
+    for_label = _('discount_form_label_coupon', default=u'Coupon')
+    for_required = _('discount_form_coupon_required',
+                     default=u'Coupon is required')
+#    for_callback = 'javascript:discount_form.autocomplete_group'
```

Comment:
So I copy pasted the class from GroupDiscountFormBase and removed the for_callback set to autocomplete groups. We only need a text field input and not autocomplete. Should I leave for_callback empty instead?